### ISSUE TITLE
Swap colors for gas & oil pipelines

### ISSUE THREAD

**Issue by Anakin86** [gravitystorm/openstreetmap-carto#4138]
At the current time gas pipelines [are rendered](https://github.com/gravitystorm/openstreetmap-carto/blob/master/style/aerialways.mss#L108) with the gray filling, and oil - with the yellow one.

I found this is perplexing. Usually gas pipelines and markers [are coloured](https://www.google.ru/search?newwindow=1&hl=ru&tbm=isch&sxsrf=ALeKk02Yhn-jp_l7EElVILYiPUZGbcsAhg%3A1588581706847&source=hp&biw=1920&bih=969&ei=StWvXsGGMYa53AP0_7GgCA&q=gas+pipeline&oq=gas+pipeline&gs_lcp=CgNpbWcQAzIECCMQJzICCAAyBAgAEB4yBAgAEB4yBAgAEB4yBAgAEB4yBAgAEB4yBAgAEB4yBAgAEB4yBAgAEB46BwgjEOoCECdQmhBY6B1gwCJoAHAAeACAAU-IAcIGkgECMTOYAQCgAQGqAQtnd3Mtd2l6LWltZ7ABCg&sclient=img&ved=0ahUKEwjB09KC6JnpAhWGHHcKHfR_DIQQ4dUDCAc&uact=5) in yellow, and on ther other side - oil is assoctiated with black color. Maybe we should swap color styles for these two substances?

**Comment by HolgerJeromin** [comment]
Moving the color to this issue for easier discussing:
gas `#c1c1c1` // Lch(78,0,0)
oil `#cfbfa5`  // Lch(78,15,83)

ref https://github.com/gravitystorm/openstreetmap-carto/pull/4070#issuecomment-599163119

> I tried to look up common colors used to identify gas vs oil pipeline, but generally they use the same symbols for flammable liquids / gases: yellow or brown.
> [...]
> I tried yellow, but this would be a little too similar to highway=secondary

**Comment by jeisenbe** [comment]
I'm ok with swapping `#c1c1c1` with `#cfbfa5` for natural gas, and then using `#c1c1c1` for oil instead, if that's considered more intuitive by most people. It would be good to get more input from other contributors who agree with that change, or oppose it. 

Or are there other colors which would work better? As mentioned in the original PR, there isn't a clear distinction between gas and oil pipelines in other symbolic systems, as far as I could find from English-language documentation.

**Comment by Adamant36** [comment]
Makes sense to me. Although, it probably shouldn't be based on some perception of how pipelines are colored in real life IMO.

**Comment by HolgerJeromin** [comment]
IMO not the pipelines are mostly colored but the markers.
At least this vendor supports the gas = yellow (-ish)
https://t-iss.com/regulation/pipe-marking/

**Comment by Anakin86** [comment]
Russian "Gazprom" also have point in [design code](https://nadymdobycha.gazprom.ru/d/textpage/5b/91/metodich_rukovodsto_gazpromdobychanadym.pdf) (with reference to national standard GOST-14202-69), that open parts of gas pipelines sholud be coloured in yellow color. See page 6 and further illustrations & photos.

For flammable liquids (as oil) brown colour is set. [The same colour code](http://tender.rosneft.ru/files/service/tt13198-08_08_2011_0.pdf) is accepted in "Rosneft". Orange colour (most similar to current OSM oil pipeline style) is reserved for pipelines with acids.

**Comment by jragusa** [comment]
For information:
* [Pipe marking on wikipedia](https://en.wikipedia.org/wiki/Pipe_marking)
* [European Union Pipe Marking Guide](http://hightechimpressions.co.za/assets/qrg_pipe-eu.pdf)
* [Plastics Industry Pipe Association of Australia Limited](http://www.pipa.com.au/wp-content/uploads/2018/10/POP203.pdf)
* [South African Standard Pipe Markers](https://d37iyw84027v1q.cloudfront.net/Common/South-African_Standard_Pipe_Markers_Brochure_English.pdf)

They suggest yellow for natural gas and brown for oil. It seems there is an international agreement about some colours

**Comment by Adamant36** [comment]
> It seems there is an international agreement about some colours

I thought the style was not suppose to represent (or copy) real life though. I'm not against it personally, but it shouldn't be the default just because that's how it is in real life if there is potentially a more intuitive way to do it. Not that I'm saying there is, But I don't think average map users know or care about the intentional color standard and it might confuse them. Maybe it would be intuitive for oil barons, but they aren't really the target audience. Although, I'm totally speculating.

**Comment by Anakin86** [comment]
> But I don't think average map users know or care about the intentional color standard and it might confuse them
I suppose that if average user doesn't know about standard, it will accpet any color on the map - it just doesn't matter for him. But it does for engineers and any person who is connected with the oil&gas industry or around it (and they are not oil barons, rather the opposite). So why not to make map more convient for maximal number of users?

**Comment by Adamant36** [comment]
> I suppose that if average user doesn't know about standard, it will accept any pipeline color scheme on the map

So then you would advocate for rendering water pipelines in brown, because it doesn't matter anyway since the average user will accept any color scheme?

**Comment by Anakin86** [comment]
>So then you would advocate for rendering water pipelines in brown, because it doesn't matter anyway since the average user will accept any color scheme?

AFAIK, there's no common international scheme for water pipelines - for example, in Germany outdoor ones are colored in blue:
<a href="https://ibb.co/DWZzFJk"><img src="https://i.ibb.co/DWZzFJk/2019-08-11-20-19-18.jpg" alt="2019-08-11-20-19-18" border="0"></a>

**Comment by Adamant36** [comment]
> AFAIK, there's no common international scheme for water pipelines - for example, in Germany outdoor ones are colored in blue:

That was the other argument I would have made for pipelines more generally. I still don't think it warrants using odd colors that are non-intuitive to most mappers though. If most people have to look up the color scheme on Wikipedia to understand what substance is in the pipe, I think that's the wrong direction to go in.

**Comment by jragusa** [comment]
I agree about blue coloured water pipelines because it's an intuitive colour for general public. But I don't think there is intuitive colours for oil and gas so we can use colour marker as guideline instead of arbitrary colours.

**Comment by jeisenbe** [comment]
> https://t-iss.com/regulation/pipe-marking/

![imagepipemarkingcolors](https://user-images.githubusercontent.com/42757252/83822731-bde5bd80-a686-11ea-840a-74209bc72d43.png)

This uses brown for fuel oil, orange for non-fuel oil, light gray for non-flammable gasses and yellow for flammable gasses. It's not clear if natural gas is considered non-flammable in this classification (from the Netherlands), anyone know? 

> For information:
> 
> * [Pipe marking on wikipedia](https://en.wikipedia.org/wiki/Pipe_marking)

There are two examples, both seem to be from the USA:
1) Hazardous materials[a] - Safety Yellow - e.g. Acetylene 
This seems to include both oil and gas?

2) Yellow for "flammables" with flash point below 38 degrees C - that doesn't include natural gas

-- | -- | -- | --
Flammables[b] & Oxidizers[c] | Safety Yellow | Black | Acetylene
-- | -- | -- | --
Combustible Fluids[d] | Safety Brown | White | Lubricating Oil

So in this case both natural gas and petroleum oil would be "safety brown"

> * [European Union Pipe Marking Guide](http://hightechimpressions.co.za/assets/qrg_pipe-eu.pdf)
-- | -- | -- | -- | --
OilS - mineral, vegetable or animal Combustible Liquids | Diesel Fuel | BrOwn | 06 c 39 | 8008
GASeS - in either gaseous or liquefied condition (except air) | Carbon Dioxide | yellOw Ochre‡ | 08 c 35 | 1017
  |   |   |   |  
Both are shades of brown, but oil is darker, and gases (including natural gas) are lighter yellow-brown.

> * [Plastics Industry Pipe Association of Australia Limited](http://www.pipa.com.au/wp-content/uploads/2018/10/POP203.pdf)

Gas is bright yellow, but there is not a color listed for oil pipelines.

> * [South African Standard Pipe Markers](https://d37iyw84027v1q.cloudfront.net/Common/South-African_Standard_Pipe_Markers_Brochure_English.pdf)

This is odd: there's "golden brown" for oil, light brown for "gases", but also light gray for "hydrocarbons". I would consider both petroleum oil and natural gas to be hydrocarbons, no? Perhaps this is meant for natural gas?

If that's correct, then the current colors match quite well. :-)

<img width="558" alt="pipeline-marker-s-africa" src="https://user-images.githubusercontent.com/42757252/83822670-8d058880-a686-11ea-8c23-e102972e0dd6.png">

**Comment by Caseyb87** [comment]
In the UK, industry practice for underground pipes/cables is:

Gas - Yellow
Oil - Black 
(and Potable (fresh/drinkable) water - Blue)

http://streetworks.org.uk/wp-content/uploads/2018/11/VOL-1-reviewed.pdf

So I support this change.

**Comment by BushmanK** [comment]
First of all, there are two major purposes of pipe color-coding:
- indoor marking for the purpose of more detailed identification by the facility personnel, maintenance personnel, first responders, building residents etc,
- outdoor marking for the purpose of more general identification by the construction workers, emergency personnel, rarely - general public.
Therefore, it makes sense to define, what purpose our mapping color scheme serves. To me, it seems like the latter one, because the vast majority of the pipelines in OSM are the outdoor pipelines.
Second, since OSM is an international project and pipelines aren't a country-specific feature, it might make sense to base the color scheme not on any of the national standards, but on ISO, specifically - ISO 14726-1. For more detailed identification within a class (say, within "fuels"), there is an extension of this standard - ISO 14726-2, that prescribes additional color bands on top of the main color. For example, oil as in "crude oil" belongs to the fuel class, which means that the main color is brown, but the extension color code for it is black. 
Here's a link to ISO 14726-2 (it covers both base and extended color schemes): https://cdn1.shipserv.com/ShipServ/pages/attachments/74678-ma/1034SS0617w.pdf

I'm not saying we have to copy the ISO standard completely, but it doesn't make a lot of sense to invent yet another completely different scheme or copy a local/national (British, Russian, American, or any other) standard.

**Comment by jeisenbe** [comment]
@BushmanK - which colors would you suggest for gas and oil pipelines, based on the linked document

It seems to be about markings for pipes on ships rather than pipelines on land. I don't see anything for natural gas - perhaps because natural gas powered ships are rare?

**Comment by BushmanK** [comment]
@jeisenbe, ISO 14726 by itself isn't just for ships or something related to the marine infrastructure.
Flammable gases (including natural gas), according to this standard, should be indicated by yellow color.
What I'm saying above is that we might consider the existing international standards (for example, by using any shade of yellow for flammable gases), I am not saying that we need to copy them.

**Comment by jeisenbe** [comment]
I see "ISO 14726-1:1999. Ships and marine technology — Identification colours for the content of piping systems" on iso.org - but this is withdrawn, superseeded by ISO 44744 - https://www.iso.org/standard/44744.html - which has the same title "Ships and marine technology — Identification colours for the content of piping systems". The pdf does not appear to be freely available, so I can't confirm if the colors are the same as in the 1999 standard.

**Comment by jeisenbe** [comment]
But this is getting off-track. The ISO specifications and other marker colors are for identifying pipes in industrial facilities. We are talking about how we ought to represent these features on a rendered map, and that adds complications.

As I mentioned in the PR (https://github.com/gravitystorm/openstreetmap-carto/pull/4070#issuecomment-599163119), there is not an established way to represent pipelines with different contents in cartography. I tried yellow for gas pipelines, but this conflicts with secondary highways (we use the yellow to red spectrum for roads). And we can't use black, because the casing is dark and the fill color needs to be lighter.

So brown and gray seemed the best next choices. I could easily swap `#c1c1c1` with `#cfbfa5`, but other colors will require further testing. 

Is there perhaps a color which will be similar to yellow, but not too similar to any of our yellow and orange road colors (https://github.com/gravitystorm/openstreetmap-carto/blob/master/style/road-colors-generated.mss):

motorway-fill: `#e892a2`
trunk-fill: `#f9b29c`
primary-fill: `#fcd6a4`
secondary-fill: `#f7fabf`
